pkg/util/containers/v2/metrics/util.go

```diff
@@ -0,0 +1,14 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package metrics
+
+import "github.com/DataDog/datadog-agent/pkg/util"
+
+func convertField(s *uint64, t **float64) {
```

Comment:
Just as a note, I also needed to convert a bunch of fields to float 64 ptr. It makes me question whether we should do this in every collector or if it would be possible to do this in an upper layer.

Reply:
@vboulineau what's your take on this? moving the conversion to the upper layer does make sense imho

Reply:
Outside of sharing the small function, I'm not sure how a generic implementation would look like? Open to ideas.

Reply:
the idea is to not let the collectors care about the conversion, and centrilize + do it once it in the check

Reply:
Okay understood, it could be interesting.
However the collectors won't be only used by the generic check, process-agent (and maybe some others, it's meant to be a generic interface).
I think it means that we'll perform 2 heaps allocation instead of one (uint64 -> *uint64 -> *float64), we also loose the ability to use the float NaN value (if the extra allocations introduced by the curent uint64 -> *float64 are too costly).